src/renderer/components/dialog/publicationInfos/opdsControls.tsx

```diff
@@ -23,37 +24,81 @@ interface IProps extends TranslatorProps, ReturnType<typeof mapDispatchToProps>,
 
 export class OpdsControls extends React.Component<IProps> {
     public render(): React.ReactElement<{}> {
-        const { publication, verifyImport, buttonIsDisabled } = this.props;
+        const { publication, buttonIsDisabled } = this.props;
         const { __ } = this.props;
 
         if (!publication) {
             return <></>;
         }
 
-        return publication.isFree ? (
-            <button
-                onClick={() => verifyImport(publication as ImportOpdsPublication)}
-                className={styles.lire}
-                disabled={buttonIsDisabled}
-            >
-                {__("catalog.addBookToLib")}
-            </button>
-        ) : publication.hasSample && (
-            <button
-                onClick={() => verifyImport(publication as ImportOpdsPublication, true)}
-                className={styles.lire}
-            >
-                {__("opds.menu.addExtract")}
-            </button>
+        return (
+            <>
+                { publication.isFree &&
+                    <button role="menuitem"
+                        onClick={ (e) => this.onAddToCatalogClick(e) }
+                        className={styles.lire}
+                        disabled={buttonIsDisabled}
+                    >
+                        {__("catalog.addBookToLib")}
+                    </button>
+                }
+                {publication.hasSample &&
+                    <button role="menuitem"
+                        onClick={(e) => this.onAddToCatalogClick(e, true)}
+                        className={styles.lire}
+                    >
+                        {__("opds.menu.addExtract")}
+                    </button>
+                }
+                {publication.buyUrl &&
+                    <button
+                        className={styles.lire}
+                    >
+                        <a role="menuitem"
+                            href={publication.buyUrl}
+                        >
+                            {__("opds.menu.goBuyBook")}
+                        </a>
+                    </button>
+                }
+                {publication.borrowUrl &&
+                    <button
+                        className={styles.lire}
+                    >
+                        <a role="menuitem"
+                            href={publication.borrowUrl}
+                        >
+                            {__("opds.menu.goLoanBook")}
+                        </a>
+                    </button>
+                }
+                {publication.subscribeUrl &&
+                    <button
+                        className={styles.lire}
+                    >
+                        <a role="menuitem"
+                            href={publication.subscribeUrl}
+                        >
+                            {__("opds.menu.goSubBook")}
+                        </a>
+                    </button>
+                }
+
+            </>
```

Comment:
The above code additions look a lot like 
https://github.com/readium/readium-desktop/pull/804/files
Is that intentional?

Reply:
I would prefer if the scope of this PR remained narrow-focused on the cover image (also, the solution proposed in 804 is slightly different). Let's see if @Scarsniik can allocate time to address the issues in the PR review, if not we will take over a branched PR from 804.

Reply:
Ok, I'll remove my works on 'opdsControl.tsx'

Becarefull for the #804 , Leo keep the ternary on 'isFree', but now we don't need it